packages/expo-device/android/src/main/java/expo/modules/device/DeviceModule.kt

```diff
@@ -0,0 +1,209 @@
+package expo.modules.device
+
+import expo.modules.core.ExportedModule
+import expo.modules.core.interfaces.RegistryLifecycleListener
+import expo.modules.core.ModuleRegistry
+import expo.modules.core.Promise
+import expo.modules.core.interfaces.ActivityProvider
+import expo.modules.core.interfaces.ExpoMethod
+import expo.modules.core.ModuleRegistryDelegate
+
+import com.facebook.device.yearclass.YearClass
+
+import android.app.Activity
+import android.app.ActivityManager
+import android.app.UiModeManager
+import android.content.Context
+import android.content.res.Configuration
+import android.os.Build
+import android.os.SystemClock
+import android.provider.Settings
+import android.view.WindowManager
+import android.util.DisplayMetrics
+
+import java.io.File
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+private const val NAME = "ExpoDevice"
+
+class DeviceModule(private val mContext: Context) : ExportedModule(mContext), RegistryLifecycleListener {
+  private val moduleRegistryDelegate: ModuleRegistryDelegate = ModuleRegistryDelegate()
+  private val mActivityProvider: ActivityProvider? by moduleRegistry()
+  private var mActivity: Activity? = null
+
+  private inline fun <reified T> moduleRegistry() = moduleRegistryDelegate.getFromModuleRegistry<T>()
+
+  // Keep this enum in sync with JavaScript
+  enum class DeviceType(val JSValue: Int) {
+    UNKNOWN(0),
+    PHONE(1),
+    TABLET(2),
+    DESKTOP(3),
+    TV(4);
+  }
+
+  override fun getName(): String {
+    return NAME
+  }
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    moduleRegistryDelegate.onCreate(moduleRegistry)
+    mActivity = mActivityProvider?.currentActivity
+  }
+
+  override fun getConstants(): Map<String, Any> = hashMapOf(
```

Comment:
Can use the more common `mapOf` to keep things simple (return type doesn't need to be a specific implementation type)

```suggestion
  override fun getConstants(): Map<String, Any> = mapOf(
```